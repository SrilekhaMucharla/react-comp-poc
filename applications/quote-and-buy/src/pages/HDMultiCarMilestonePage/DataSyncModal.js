import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row } from 'react-bootstrap';
import { AnalyticsHDModal as HDModal } from '../../web-analytics';
import * as messages from './HDMultiCarMilestonePage.messages';
import { pageMetadataPropTypes } from '../../constant/propTypes';

const DataSyncModal = ({
    onCancel, onConfirm, onClose, show, pageMetadata, hideClose, vehicleDetails, vehicleName
}) => (
    <HDModal
        webAnalyticsView={{ ...pageMetadata, page_section: messages.areYouSure }}
        webAnalyticsEvent={{ event_action: messages.areYouSure }}
        customStyle="rev-button-order footer-btns-w-100"
        id="data-sync-modal"
        show={show}
        headerText={messages.areYouSure}
        cancelLabel={messages.goBack}
        confirmLabel={messages.yesImSure}
        onCancel={onCancel}
        onConfirm={onConfirm}
        onClose={onClose}
        hideClose={hideClose}
    >
        <p className="mb-4 subHeaderText">{messages.subHeaderText}</p>
        <Row>
            <Col>
                <div className="completed-info-card__main-text car">{vehicleDetails}</div>
            </Col>
        </Row>
        <Row>
            <Col>
                <div className="completed-info-card__additional-text mt-1">{vehicleName}</div>
            </Col>
        </Row>
        <p className="mb-4 mt-4 innerParagraphText">{messages.innerParagraphText}</p>
    </HDModal>
);

DataSyncModal.propTypes = {
    onCancel: PropTypes.func.isRequired,
    onConfirm: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired,
    show: PropTypes.bool,
    hideClose: PropTypes.bool,
    pageMetadata: PropTypes.shape(pageMetadataPropTypes).isRequired,
    vehicleDetails: PropTypes.string.isRequired,
    vehicleName: PropTypes.string.isRequired,
};

DataSyncModal.defaultProps = {
    show: false,
    hideClose: false
};

export default DataSyncModal;
