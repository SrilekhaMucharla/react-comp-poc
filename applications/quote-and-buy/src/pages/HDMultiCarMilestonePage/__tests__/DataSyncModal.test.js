import React from 'react';
import { shallow } from 'enzyme';
import DataSyncModal from '../DataSyncModal';

describe('DataSyncModal', () => {
    // given
    const onCancel = jest.fn();
    const onConfirm = jest.fn();
    const onClose = jest.fn();
    const syncSaveChanges = 'This will save any changes made to:';
    const id = 'data-sync-modal';
    const innerParagraphText = 'You won\'t be able to cancel these changes once they\'ve been saved.';
    const vehicleDetails = 'AV12BGE';
    const vehicleName = 'Test';

    const getWrapper = (props = {}) => shallow(
        <DataSyncModal
            onCancel={onCancel}
            onConfirm={onConfirm}
            onClose={onClose}
            syncSaveChanges={syncSaveChanges}
            innerParagraphText={innerParagraphText}
            vehicleDetails={vehicleDetails}
            vehicleName={vehicleName}
            show
            {...props} />
    );

    it('should render correctly and match the snapshot', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper).toMatchSnapshot();
    });

    it('should display HDModal', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper.find(`#${id}`)).toHaveLength(1);
    });

    it('should contain syncSaveChanges', () => {
        // given
        const wrapper = getWrapper();
        console.log(wrapper.debug());
        // then
        expect(wrapper.find('p').at(0).text()).toContain(syncSaveChanges);
    });

    it('should contain innerParagraphText', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper.find('p').at(1).text()).toContain(innerParagraphText);
    });

    it('should contain vehicleDetails', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper.find('div').at(0).text()).toContain(vehicleDetails);
    });

    it('should contain vehicleName', () => {
        // given
        const wrapper = getWrapper();
        // then
        expect(wrapper.find('div').at(1).text()).toContain(vehicleName);
    });

    it('should pass functions to HDModal', () => {
        // given
        const wrapper = getWrapper();
        const modalProps = wrapper.find(`#${id}`).props();
        // then
        expect(modalProps.onCancel).toBe(onCancel);
        expect(modalProps.onClose).toBe(onClose);
        expect(modalProps.onConfirm).toBe(onConfirm);
    });
});
