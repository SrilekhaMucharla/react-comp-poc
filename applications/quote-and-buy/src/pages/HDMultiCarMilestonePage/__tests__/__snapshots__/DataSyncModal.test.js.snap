// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSyncModal should render correctly and match the snapshot 1`] = `
<ForwardRef
  cancelLabel="No, go back"
  confirmLabel="Yes, i'm sure"
  customStyle="rev-button-order footer-btns-w-100"
  headerText="Are you sure?"
  hideClose={false}
  id="data-sync-modal"
  onBeforeOpen={null}
  onCancel={[MockFunction]}
  onClose={[MockFunction]}
  onConfirm={[MockFunction]}
  show={true}
  webAnalyticsEvent={
    Object {
      "event_action": "Are you sure?",
    }
  }
  webAnalyticsView={
    Object {
      "page_section": "Are you sure?",
    }
  }
>
  <p
    className="mb-4 subHeaderText"
  >
    This will save any changes made to:
  </p>
  <Row
    noGutters={false}
  >
    <Col>
      <div
        className="completed-info-card__main-text car"
      >
        AV12BGE
      </div>
    </Col>
  </Row>
  <Row
    noGutters={false}
  >
    <Col>
      <div
        className="completed-info-card__additional-text mt-1"
      >
        Test
      </div>
    </Col>
  </Row>
  <p
    className="mb-4 mt-4 innerParagraphText"
  >
    You won't be able to cancel these changes once they've been saved.
  </p>
</ForwardRef>
`;
